internal/utils_fs.go

```diff
@@ -13,24 +13,41 @@ import (
 
 // getRandomDirectory creates a random directory in /tmp, creates the directories and returns the full path
 // directory is of the form `/tmp/<random-word>/<random-word>/<random-word>`
-func getRandomDirectory(stageHarness *test_case_harness.TestCaseHarness) (string, error) {
+// If performCleanup is true, the directory will be cleaned up when the test is completed
+// The total possible directories is 10^3 = 1000
+// This can be used without cleanup in most cases
+func getRandomDirectory(stageHarness *test_case_harness.TestCaseHarness, performCleanup bool) (string, error) {
 	randomDir := path.Join("/tmp", random.RandomWord(), random.RandomWord(), random.RandomWord())
-	if err := os.MkdirAll(randomDir, 0755); err != nil {
-		return "", fmt.Errorf("CodeCrafters internal error. Error creating directory %s: %v", randomDir, err)
+	for {
+		if _, err := os.Stat(randomDir); os.IsNotExist(err) {
+			if err := os.MkdirAll(randomDir, 0755); err != nil {
+				return "", fmt.Errorf("CodeCrafters internal error. Error creating directory %s: %v", randomDir, err)
+			}
+			break
+		}
+		randomDir = path.Join("/tmp", random.RandomWord(), random.RandomWord(), random.RandomWord())
 	}
 
-	// Automatically cleanup the directory when the test is completed
-	stageHarness.RegisterTeardownFunc(func() {
-		grandParentDir := path.Dir(path.Dir(randomDir))
-		cleanupDirectories([]string{grandParentDir})
-	})
+	// Automatically cleanup the directory when the test is completed, if requested
+	if performCleanup {
+		stageHarness.RegisterTeardownFunc(func() {
+			grandParentDir := path.Dir(path.Dir(randomDir))
+			cleanupDirectories([]string{grandParentDir})
+		})
+	}
 
 	return randomDir, nil
 }
 
-// getShortRandomDirectory creates a random directory in /tmp, creates the directories and returns the full path
+func getRandomDirectoryWithCleanup(stageHarness *test_case_harness.TestCaseHarness) (string, error) {
```

Comment:
@ryan-gang assuming we **did** need a method with and without cleanup, I'd name them the reverse: `getRandomDirectory` and `getRandomDirectoryWithoutCleanup`. 

Reason: In most cases we do want to cleanup random dirs, so make it easy to not mistakenly forget. In the case where we don't want to cleanup, we want to be specific about it - so let's force users to say `WithoutCleanup` to enforce that thinking